tests/rptest/tests/datalake/schema_evolution_test.py

```diff
@@ -20,210 +29,560 @@
-            "type": "string"
-        },
-        {
-            "name": "ordinal",
-            "type": "int"
-        }
-    ]
-}
-"""
-
-legal_evo_schema_avro = """
-{
-    "type": "record",
-    "name": "VerifierRecord",
-    "fields": [
-        {
-            "name": "verifier_string",
-            "type": "string"
-        },
-        {
-            "name": "ordinal",
-            "type": "long"
-        },
-        {
-            "name": "another",
-            "type": "float"
-        }
-    ]
-}
-"""
-
-legal_evo_schema_avro_2 = """
-{
-    "type": "record",
-    "name": "VerifierRecord",
-    "fields": [
-        {
-            "name": "verifier_string",
-            "type": "string"
-        },
-        {
-            "name": "ordinal",
-            "type": "long"
-        }
-    ]
-}
-"""
-
-illegal_evo_schema_avro = """
-{
-    "type": "record",
-    "name": "VerifierRecord",
-    "fields": [
-        {
-            "name": "verifier_string",
-            "type": "double"
-        },
-        {
-            "name": "ordinal",
-            "type": "long"
+from ducktape.errors import TimeoutError
+
+QUERY_ENGINES = [
+    QueryEngineType.SPARK,
+    QueryEngineType.TRINO,
+]
+
+
+class AvroSchema:
+    TableDescription = list[tuple[str, str]]
+
+    def __init__(self,
+                 fields: list[dict],
+                 generate_record: Callable[[float], dict],
+                 spark_table: TableDescription = [],
+                 trino_table: TableDescription = [],
+                 name="AvroRecord"):
+        self._rep: dict = {
+            "type": "record",
+            "name": name,
+            "fields": fields,
         }
-    ]
-}
-"""
-
-
-class SchemaEvolutionTest(RedpandaTest):
-    def __init__(self, test_context):
-        super(SchemaEvolutionTest, self).__init__(
-            test_context=test_context,
-            num_brokers=1,
-            si_settings=SISettings(test_context,
-                                   cloud_storage_enable_remote_read=False,
-                                   cloud_storage_enable_remote_write=False),
-            extra_rp_conf={
-                "iceberg_enabled": True,
-                "iceberg_catalog_commit_interval_ms": 5000,
+        self._generate_record = generate_record
+        self._spark_table = spark_table
+        self._trino_table = trino_table
+
+    def load(self):
+        return avro.loads(json.dumps(self._rep))
+
+    def table(self, engine: QueryEngineType):
+        return self._spark_table if engine == QueryEngineType.SPARK else self._trino_table
+
+    @property
+    def field_names(self) -> list[str]:
+        return [field['name'] for field in self._rep['fields']]
+
+    @property
+    def fields(self) -> list[dict]:
+        return self._rep['fields']
+
+    def gen(self, x: float) -> dict:
+        return self._generate_record(x)
+
+
+class EvolutionTestCase(NamedTuple):
+    initial_schema: AvroSchema
+    next_schema: AvroSchema
+
+
+LEGAL_TEST_CASES = {
```

Comment:
This is so great thank you I think its' way easier to grok when everything is colocated together ❤️